mailpoet/assets/js/src/automation/integrations/core/steps/delay/types/delayTypes.ts

```diff
@@ -0,0 +1,26 @@
+import { SelectControl } from '@wordpress/components';
+
+export type DelayTypes = SelectControl.Option & {
+  singular: string;
+  plural: string;
+};
+export const DelayTypeOptions: DelayTypes[] = [
+  {
+    label: 'Hours',
+    singular: 'hour',
+    plural: 'hours',
+    value: 'HOURS',
+  },
+  {
+    label: 'Days',
+    singular: 'day',
+    plural: 'days',
+    value: 'DAYS',
+  },
+  {
+    label: 'Weeks',
+    singular: 'week',
+    plural: 'weeks',
+    value: 'WEEKS',
+  },
+];
```

Comment:
We didn't solve translations yet, so this may just be a reflection for the future, but I'm thinking about making these translatable... Do you think we could use the [Intl API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat) instead of translating all these forms (singular, plural, etc.)? After all, this won't work for Czech, for instance, as we have multiple plurals.

Reply:
:thinking: 
Did look into the Intl API, but not sure how the API could help us here. I propose for now to change the DateType object and add a `subtitle(value:number) => string` callback function there. Once we come to internationalization, we can pick it up from there I guess. [ced0355](https://github.com/mailpoet/mailpoet/pull/4275/commits/ced0355362c9dfbe22bf005c157f247da9e3076f)

Reply:
@websupporter What I meant is the number format. Try this in your browser console:

```js
(5).toLocaleString('de', {
  style: 'unit',
  unit: 'second',
  unitDisplay: 'long',
});
```

If we pass the language from MailPoet then this would solve it nicely, I guess. What do you think?

Reply:
I hesitate currently. As of know, we would need to hard code `'en'` anyway, because otherwise we get an English string with Czech words in it.

What I like about it, we can reduce now the "knowledge" of the `DelayType` to just the JavaScript unit and we do not need to move all texts into the `DelayType`. What I worry about is: I am not sure if there are strings in some languages where `5 hours` might be separated. What I mean: "Wait for `5 hours`" might correspond in another language to  a string like "The amount of `hours` you have to wait is `5`"

With using a JavaScript function we delegate a part of the translation to the domain of the browser and invite inconsistencies between browsers and other issues (Chrome does not yet support `locale X`, Firefox `locale Y`), we do not want to have and `_n()` could save us the headache I suppose.

I would suggest to return to this topic, when we have decided on translations @JanJakes

Reply:
@websupporter OK, let's keep it as it is now. The `Intl` API is amazing if you need to display the time unit alone (without being embedded in a sentence) or a relative date for comments, etc. but it's true we have a slightly different case.